### DISCUSSION TITLE
Injected dependency in custom Kafka Interceptor

### DISCUSSION THREAD

**Discussion by mariuszbozek** [quarkusio/quarkus#28547] [Q&A]
Hi,

Is there an option to inject dependency into a custom Kafka Producer/Consumer Interceptor?

an example
```
public class CustomProducerInterceptor<K, V> implements ProducerInterceptor<K, V> {

  private static final Logger LOGGER = LoggerFactory.getLogger(CustomProducerInterceptor.class);

  @Inject
  CustomService service;
  ...
}
```
service (create with @Singleton) always return null

[a comment by an automated account is not shown]

**Comment by alesj** [comment]
> Is there an option to inject dependency into a custom Kafka Producer/Consumer Interceptor?

I doubt it, since interceptor instantiation is deep in the Kafka layer, which is not tied with CDI ...

**Reply by cescoffier** [reply]
Interceptors are managed by Kafka itself and are instantiated by the Kafka client. So, you cannot inject beans.
I would recommend using programmatic lookups as exexplained in https://dzone.com/refcardz/contexts-and-depencency#section-12.
